packages/common/src/content/_internal.ts

```diff
@@ -347,23 +347,60 @@ export const getAdditionalResources = (
   metadata?: any,
   requestOptions?: IHubRequestOptions
 ): IHubAdditionalResource[] => {
-  let rawResources: IAGOAdditionalResource | IAGOAdditionalResource[] =
-    getProp(metadata, "metadata.distInfo.distTranOps.onLineSrc") || null;
-
-  if (rawResources === null) {
-    return null;
-  }
+  const rawResources: IAGOAdditionalResource[] = extractRawResources(metadata);
+  return (
+    rawResources &&
+    rawResources.map(
+      (resource: IAGOAdditionalResource): IHubAdditionalResource => ({
+        name: resource.orName,
+        url: getAdditionalResourceUrl(resource, item, requestOptions),
+        isDataSource: isDataSourceOfItem(resource, item),
+      })
+    )
+  );
+};
 
-  // Coerce to array since rawResources will be an object if only 1 resource is available
-  rawResources = Array.isArray(rawResources) ? rawResources : [rawResources];
+/**
+ * Extracts additional resources from formal item metadata.
+ * If none are available, null is returned.
+ *
+ * @param metadata the formal item metadata
+ * @returns an array of all additional resources, or null
+ */
+export const extractRawResources = (
+  metadata?: any
+): IAGOAdditionalResource[] => {
+  const rawResources: IAGOAdditionalResource[] = [];
 
-  return rawResources.map(
-    (resource: IAGOAdditionalResource): IHubAdditionalResource => ({
-      name: resource.orName,
-      url: getAdditionalResourceUrl(resource, item, requestOptions),
-      isDataSource: isDataSourceOfItem(resource, item),
-    })
+  // The property path to additional resources should be fairly simple.
+  // In many cases, it's just `metadata.metadata.distInfo.distTranOps.onLineSrc`.
+  // However, since `distInfo`, `distTranOps` and `onLineSrc` can be either
+  // Objects OR Arrays, we have to do all this looping.
+  maybeCastToArray(getProp(metadata, "metadata.distInfo")).forEach(
```

Comment:
@drspacemanphd this is my interpretation of your indexer commit, without using lodash. Would you mind double checking it?

Reply:
Yes that is right. "Officially", you can only make `distTranOps` and `onLineSrc` arrays via the metadata editor, but scripting shows that AGO will not invalidate `distInfo` arrays

Reply:
Also if `getProp` allows, this would be where I would put a default value of an empty array in case `distInfo` does not exist.